### PR TITLE
Added last processing time for item and test session.

### PR DESCRIPTION
**Pull request by julien-sebire**
This PR aims at adding a timestamp (with microseconds) for each ended attempt, in order to give some more fine-grain feedback when returning test result.
Result about each item is return with the last processing time for this item, and not only at the time of processing the whole test result.

### CODE REVIEW

File: src/qtism/runtime/tests/lastProcessingTimeAwareTrait.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * @author Julien Sébire <julien@taotesting.com>
+ * @license GPLv2
+ */
+
+namespace qtism\runtime\tests;
+
+use DateTime;
+use DateTimeZone;
+
+trait lastProcessingTimeAwareTrait
```
Comment (bugalot) [inline src/qtism/runtime/tests/lastProcessingTimeAwareTrait.php]: Shouldn't we use a Capital letter for the first character? Like `LastProcessingTimeAwareTrait`

File: test/qtismtest/runtime/tests/AssessmentItemSessionTest.php [hunk 2]
```diff
@@ -106,6 +108,13 @@ public function testEvolutionBasic()
         $this->assertTrue($itemSession->isResponded());
         $this->assertTrue($itemSession->isResponded(false));
         
+        // Checks the last processing time.
+        $lastProcessingTime = $itemSession->getLastProcessingTime();
+        $this->assertInstanceOf(DateTime::class, $lastProcessingTime);
+        // Checks that the DateTime is less that one second in the past.
+        $difference = (new DateTime('now', new DateTimeZone('UTC')))->diff($lastProcessingTime);
```
Comment (bugalot) [inline test/qtismtest/runtime/tests/AssessmentItemSessionTest.php]: I would not rely on time in a CI context.

File: test/qtismtest/runtime/tests/AssessmentTestSessionTest.php [hunk 3]
```diff
@@ -193,7 +195,14 @@ public function testLinearSkipAll()
 	    $this->assertFalse($assessmentTestSession->getCurrentTestPart());
 	    $this->assertFalse($assessmentTestSession->getCurrentNavigationMode());
 	    $this->assertFalse($assessmentTestSession->getCurrentSubmissionMode());
-	}
+	    
+        // Checks the last processing time.
+        $lastProcessingTime = $assessmentTestSession->getLastProcessingTime();
+        $this->assertInstanceOf(DateTime::class, $lastProcessingTime);
+        // Checks that the DateTime is less that one second in the past.
+        $difference = (new DateTime('now', new DateTimeZone('UTC')))->diff($lastProcessingTime);
```
Comment (bugalot) [inline test/qtismtest/runtime/tests/AssessmentTestSessionTest.php]: I would not rely on time in a CI context.

File: test/qtismtest/runtime/tests/lastProcessingTimeAwareTraitTest.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    use lastProcessingTimeAwareTrait;
+    
+    public function test()
+    {
+        $this->updateLastProcessingTime();
+        $lastProcessingTime = $this->getLastProcessingTime();
+        $this->assertInstanceOf(DateTime::class, $lastProcessingTime);
+        
+        // Checks that the DateTime is less that one second in the past.
+        $difference = (new DateTime('now', new DateTimeZone('UTC')))->diff($lastProcessingTime);
```
Comment (bugalot) [inline test/qtismtest/runtime/tests/lastProcessingTimeAwareTraitTest.php]: I would not rely on time in a CI context.

**Comment by julien-sebire** [comment]
Closing as not needed anymore it seems (almost 3 years old).

**Pull request closed without merging**